### PR TITLE
docs(dropdown): implement and document attrs and vars

### PR DESCRIPTION
**Pull request by lalithkarikelli**
## Description

`hx-dropdownSelect` : Implement and document attributes and variables

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF 2105

## Before you request a review for this PR:

- [x] For UI changes, did you manually test in recent versions of modern browsers (Chrome, Firefox, and Safari)?
- [x] For UI changes, did you manually test in IE11 and legacy Edge?
- [ ] Did you add component tests for any new code?
- [ ] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [ ] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: src/scss/components/dropdown-select/mixins/SelectFacade.scss [hunk 1]
```diff
@@ -7,7 +7,7 @@
   }
   @else {
     border: 1px solid $gray-500;
```
Comment (BY00565233) [inline src/scss/components/dropdown-select/mixins/SelectFacade.scss:9]: please change to border: 1px solid var(--hxDropdown-border, $gray-500);

File: src/scss/components/dropdown-select/mixins/SelectFacade.scss [hunk 2]
```diff
@@ -18,7 +18,7 @@
   }
   @else {
     border: 2px solid $red-900;
```
Comment (BY00565233) [inline src/scss/components/dropdown-select/mixins/SelectFacade.scss:20]: please follow same pattern remaining border properties
change it to border: 2px solid var(--hxDropdown-invalid-border, $red-900);

File: src/scss/components/dropdown-select/mixins/SelectFacade.scss (hunk 1 above)
Comment (BY00565233) [inline src/scss/components/dropdown-select/mixins/SelectFacade.scss:9]: please change to border: 1px solid var(--hxDropdown-border, $gray-500);

File: src/scss/components/dropdown-select/mixins/SelectFacade.scss (hunk 2 above)
Comment (BY00565233) [inline src/scss/components/dropdown-select/mixins/SelectFacade.scss:20]: please follow same pattern through out the application
change to border: 2px solid var(--hxDropdown-invalid-border, $gray-900);

File: src/scss/components/dropdown-select/mixins/SelectFacade.scss (hunk 1 above)
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/mixins/SelectFacade.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/mixins/SelectFacade.scss (hunk 2 above)
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/mixins/SelectFacade.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/mixins/SelectControl.scss [hunk 3]
```diff
@@ -8,10 +8,9 @@
       color: transparent;
       text-shadow: 0 0 0 $gray-900;
     }
-  }
-  @else {
-    background-color: $gray-0;
-    color: $gray-900;
+  } @else {
+    background-color: var(--hxDropdown-unfocused-backgroundColor, $gray-0);
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/mixins/SelectControl.scss:12]: Changes have been done, please review.

File: src/scss/components/dropdown-select/mixins/SelectControl.scss [hunk 4]
```diff
@@ -23,10 +22,9 @@
       color: transparent;
       text-shadow: 0 0 0 $gray-900;
     }
-  }
-  @else {
-    background-color: $gray-0;
-    color: $gray-900;
+  } @else {
+    background-color: var(--hxDropdown-unfocused-invalid-backgroundColor, $gray-0);
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/mixins/SelectControl.scss:26]: Changes have been done, please review.

File: src/scss/components/dropdown-select/mixins/SelectControl.scss [hunk 5]
```diff
@@ -38,10 +36,9 @@
       color: transparent;
       text-shadow: 0 0 0 $gray-900;
     }
-  }
-  @else {
-    background-color: $gray-100;
-    color: $gray-700;
+  } @else {
+    background-color: var(--hxDropdown-unfocused-disabled-backgroundColor, $gray-100);
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/mixins/SelectControl.scss:40]: Changes have been done, please review.

**Review by BY00565233 — APPROVED** [review]
DEV LGTM

**Review by 100stacks — CHANGES_REQUESTED** [review]
@lalithkarikelli, please squash commits.

**Comment by lalithkarikelli** [comment]
> @lalithkarikelli, please squash commits.

@100stacks rebased and squashed the commits

**Review by 100stacks — CHANGES_REQUESTED** [review]
### Please rebase.

**Comment by lalithkarikelli** [comment]
> ### Please rebase.

done.

File: src/scss/components/dropdown-select/mixins/SelectControl.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (100stacks) [reply src/scss/components/dropdown-select/mixins/SelectControl.scss:12]: I'm not seeing the recommended variable name change, please double check.

File: src/scss/components/dropdown-select/mixins/SelectControl.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (100stacks) [reply src/scss/components/dropdown-select/mixins/SelectControl.scss:26]: I'm not seeing the recommended variable name change, please double check.

File: src/scss/components/dropdown-select/mixins/SelectControl.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (100stacks) [reply src/scss/components/dropdown-select/mixins/SelectControl.scss:40]: I'm not seeing the recommended variable name change, please double check.

File: src/scss/components/dropdown-select/mixins/SelectFacade.scss [hunk 6]
```diff
@@ -2,40 +2,36 @@
 
 @mixin __selectFacade($focused: false) {
   @if $focused {
-    border: 1px solid $cyan-700;
+    border: 1px solid var(--hxDropdown-focused-borderColor, $cyan-700);
```
Comment (100stacks) [inline src/scss/components/dropdown-select/mixins/SelectFacade.scss:5]: Please update to the new Color Palette 2.0 color variables.

File: src/scss/components/dropdown-select/mixins/SelectFacade.scss [hunk 7]
```diff
@@ -2,40 +2,36 @@
...
+  } @else {
+    border: 1px solid var(--hxDropdown-unfocused-borderColor, $gray-500);
+    color: var(--hxDropdown-color, $gray-900);
   }
 }
 
 @mixin __selectFacade--invalid($focused: false) {
   @if $focused {
-    border: 2px solid $red-900;
+    border: 2px solid var(--hxDropdown-invalid-focused-borderColor, $red-900);
```
Comment (100stacks) [inline src/scss/components/dropdown-select/mixins/SelectFacade.scss:15]: Please update to the new Color Palette 2.0 color variables.

File: src/scss/components/dropdown-select/mixins/SelectFacade.scss [hunk 8]
```diff
@@ -2,40 +2,36 @@
...
     box-shadow: $focus-glow-invalid;
-  }
-  @else {
-    border: 2px solid $red-900;
-    color: $gray-900;
+  } @else {
+    border: 2px solid var(--hxDropdown-invalid-unfocused-borderColor, $red-900);
```
Comment (100stacks) [inline src/scss/components/dropdown-select/mixins/SelectFacade.scss:18]: Please update to the new Color Palette 2.0 color variables.

**Review by 100stacks — CHANGES_REQUESTED** [review]

File: src/scss/components/dropdown-select/mixins/SelectFacade.scss (hunk 6 above)
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/mixins/SelectFacade.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/mixins/SelectFacade.scss (hunk 7 above)
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/mixins/SelectFacade.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/mixins/SelectFacade.scss (hunk 8 above)
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/mixins/SelectFacade.scss]: Changes have been done, please review.

**Review by 100stacks — APPROVED** [review]
Dev LGTM

**Pull request merged**